libnetwork/sandbox_dns_unix.go

```diff
@@ -27,6 +29,35 @@ const (
 	resolverIPSandbox = "127.0.0.11"
 )
 
+// ctrTaskCreatedDNS is to be called after the container task has been created,
+// before it is started. Sandbox.osSbox has been set up at this point, and can be
+// used to probe for IPv6 being enabled in the container.
+func (sb *Sandbox) ctrTaskCreatedDNS() error {
+	sb.mu.Lock()
+	osSbox := sb.osSbox
+	sb.mu.Unlock()
+
+	// Set Sandbox.ipv6Disabled according to whether the container's 'lo' interface
+	// has an IPv6 address ('::1' is assigned automatically, even when the container
+	// has no IPv6-enabled endpoints). There should be an osSbox, but default to
+	// ipv6Disabled=false if not.
+	if osSbox == nil {
```

Comment:
In which circumstances will `osSbox == nil`? Does that include `--net=host`? If so, should the host netns should be probed for IPv6-ness?

Reply:
I haven't found a case where the osSbox is `nil`. But, because its presence depends on its caller having called `withLibnetwork()` so that the SetKey stuff happens, it seemed best to be defensive rather than assume it's set and panic (or bail out) if it's not.

(Hence the comment "There should be an osSbox, but ...", and the `Warnf()`.)

Reply:
`--net=none`, then. That's not an exceptional case, so we shouldn't fill the daemon log with noise.
https://github.com/moby/moby/blob/df3a3211645c8b9406a74f97777c45e7b221d127/daemon/oci_linux.go#L64-L68

Reply:
This has gone-away now I've moved all the action into SetKey().

Reply:
That code came back, along with the `Warnf`, because despite appearances in the code you highlighted - SetKey is called with an osSbox for a `--net=none`.

I've investigated now - `c.Config.NetworkDisabled` is only set for the default bridge during daemon startup, if the daemon's started with `--bridge none`.

So, the right thing to do here is just `return nil` ... I'll do that.